### ISSUE TITLE
Cannot Use GpuDelegate on simple model with 3D input

### ISSUE THREAD

**Issue by ghiles-bgp** [tensorflow/tensorflow#34525]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS Catalina and Android 10
- TensorFlow installed from (source or binary): binary
- TensorFlow version (or github SHA if from source): 2.0.0

**Provide the text output from tflite_convert**

```
2019-11-22 17:05:40.034915: I tensorflow/core/grappler/devices.cc:60] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0 (Note: TensorFlow was not compiled with CUDA support)
2019-11-22 17:05:40.034972: I tensorflow/core/grappler/clusters/single_machine.cc:356] Starting new session
2019-11-22 17:05:40.046101: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:716] Optimization results for grappler item: graph_to_optimize
2019-11-22 17:05:40.046128: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:718]   function_optimizer: Graph size after: 39 nodes (34), 63 edges (58), time = 4.433ms.
2019-11-22 17:05:40.046137: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:718]   function_optimizer: function_optimizer did nothing. time = 0.069ms.
2019-11-22 17:05:40.075988: I tensorflow/core/grappler/devices.cc:60] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0 (Note: TensorFlow was not compiled with CUDA support)
2019-11-22 17:05:40.076066: I tensorflow/core/grappler/clusters/single_machine.cc:356] Starting new session
2019-11-22 17:05:40.089383: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:716] Optimization results for grappler item: graph_to_optimize
2019-11-22 17:05:40.089403: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:718]   constant folding: Graph size after: 34 nodes (-5), 54 edges (-9), time = 9.559ms.
2019-11-22 17:05:40.089408: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:718]   constant folding: Graph size after: 34 nodes (0), 54 edges (0), time = 0.949ms.
```

**Any other info / logs**

*The model*

```
import tensorflow as tf

model = tf.keras.models.Sequential()
model.add(tf.keras.layers.Input(shape=(40, 300), dtype=tf.dtypes.float32))
model.add(tf.keras.layers.Dense(10))

saved_model_dir = './saved_model/'
model.save(saved_model_dir, save_format='tf')
```

*TFLite conversion code*

```
tflite_model_path = './model.tflite'
converter = tf.lite.TFLiteConverter.from_saved_model(saved_model_dir)
converter.target_spec.supported_ops = [tf.lite.OpsSet.TFLITE_BUILTINS]
tflite_model = converter.convert()
with open(tflite_model_path, 'wb') as f:
    f.write(tflite_model)
```

*Android code*

```
String dst = "<Absolute path to the .tflite file>"
Interpreter.Options options = new Interpreter.Options();
GpuDelegate gpuDelegate = new GpuDelegate();
options.addDelegate(gpuDelegate);
Interpreter interpreter = new Interpreter(new File(dst), options);  // The exception is thrown here

float[][][] input = new float[1][40][300];
for (int i = 0; i < input.length; i++) {
    for (int j = 0; j < input[i].length; j++) {
        for (int k = 0; k < input[i][j].length; k++) {
            input[i][j][k] = (float) Math.random();
        }
    }
}
float[][][] output = new float[1][40][10];
interpreter.run(input, output);
```

This works on Android when running on CPU (not using the GPU delegate), but when using the GPU delegate, the following exception is thrown:
```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.tflitegpu, PID: 24680
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.example.tflitegpu/com.example.tflitegpu.MainActivity}: java.lang.IllegalArgumentException: Internal error: Failed to apply delegate: TfLiteGpuDelegate Init: FULLY_CONNECTED: Amount of input data should match weights width
    TfLiteGpuDelegate Prepare: delegate is not initialized
    Node number 2 (TfLiteGpuDelegateV2) failed to prepare.
```

The Full Traceback and the TFLite file are attached.

Thank you for your help!

[traceback.txt](https://github.com/tensorflow/tensorflow/files/3880354/traceback.txt)
[model.tflite.zip](https://github.com/tensorflow/tensorflow/files/3880355/model.tflite.zip)

**Comment by terryheo** [comment]
@ghiles-bgp , was your intention making fully connected layer with 40 x 300 input with 1 batch?

The error happens at https://github.com/tensorflow/tensorflow/blob/master/tensorflow/lite/delegates/gpu/common/model_builder.cc#L1206 since input shape is 40 x 300 and batch size is 1 while width of weights is 300. @impjdi , could you confirm the current GPU delegate handles multi dimensional input?

**Comment by impjdi** [comment]
Thanks for looking into this @terryheo 

IIRC I have received an issue related to

> "Amount of input data should match weights width"

@ekaterinaignasheva Was that assigned to you?

**Comment by impjdi** [comment]
I checked with @ekaterinaignasheva in person.

`FULLY_CONNECTED` is essentially matrix matrix multiplication, and we unfortunately don't support it, only matrix vector multiplication.  We're currently short staffed and can't prioritize to implement that.  Would it be possible to rewrite your graph to use 1x1 convolution instead?

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by ghiles-bgp** [comment]
Thank you for the 1x1 convolution suggestion, we were not aware of the trick. Since the matrix by matrix multiplication is not supported I think the issue can be closed since this is more a feature request and you seem to be aware of it.
Thanks again for you help!
